pkg/vm/engine/tae/moengine/txn_wrapper.go

```diff
@@ -45,13 +46,11 @@ func (w *wrappedEngine) New(options ...client.TxnOption) (client.TxnOperator, er
 }
 
 func (w *wrappedEngine) NewWithSnapshot(snapshot []byte) (client.TxnOperator, error) {
-	tx, err := w.engine.StartTxn(snapshot)
-	if err != nil {
-		return nil, err
+	txn := w.engine.(*txnEngine).impl.TxnMgr.GetTxnByCtx(snapshot)
+	if txn == nil {
+		return nil, data.ErrNotFound
```

Comment:
the paradiagm, data.ErrNotFound, I think is a very bad practice.    It may save a few key strokes (maybe not), but it lost all the context.   Please switch to moerr.NewError

Reply:
done